Common/Helpers/AzureTableStorageClient.cs

```diff
@@ -10,37 +11,63 @@
 
 namespace Microsoft.Azure.Devices.Applications.RemoteMonitoring.Common.Helpers
 {
-    public class AzureTableStorageHelper : IAzureTableStorageHelper
+    public class AzureTableStorageClient : IAzureTableStorageClient
     {
-        private readonly string _storageConnectionString;
-        private readonly string _tableName;
         private CloudTable _table;
+        private readonly CloudTableClient _tableClient;
+        private readonly string _tableName;
 
-        public AzureTableStorageHelper(string storageConnectionString, string tableName)
+        public AzureTableStorageClient(string storageConnectionString, string tableName)
         {
-            _storageConnectionString = storageConnectionString;
+            CloudStorageAccount storageAccount = CloudStorageAccount.Parse(storageConnectionString);
+            _tableClient = storageAccount.CreateCloudTableClient();
             _tableName = tableName;
-            CloudStorageAccount storageAccount = CloudStorageAccount.Parse(_storageConnectionString);
-            CloudTableClient tableClient = storageAccount.CreateCloudTableClient();
-            _table = tableClient.GetTableReference(_tableName);
         }
-
-        public async Task<CloudTable> GetTableAsync()
+        private async Task<CloudTable> GetCloudTableAsync()
         {
-            await _table.CreateIfNotExistsAsync();
+            if (_table == null && _tableName != null)
+            {
+                _table = _tableClient.GetTableReference(_tableName);
+                await _table.CreateIfNotExistsAsync();
+            }
             return _table;
         }
-
-        public CloudTable GetTable()
+        private CloudTable GetCloudTable()
         {
+            if (_table != null)
+            {
+                return _table;
+            }
+            _table = _tableClient.GetTableReference(_tableName);
             _table.CreateIfNotExists();
             return _table;
         }
 
+        public TableResult Execute(TableOperation tableOperation)
+        {
+            CloudTable table = this.GetCloudTable();
+            return table.Execute(tableOperation);
+        }
+        public async Task<TableResult> ExecuteAsync(TableOperation operation)
+        {
+            CloudTable table = await this.GetCloudTableAsync();
+            return await table.ExecuteAsync(operation);
+        }
+        public IEnumerable<T> ExecuteQuery<T>(TableQuery<T> tableQuery) where T : TableEntity, new()
```

Comment:
T must be a non-abstract type with a public parameterless constructor, so cannot use ITableEntity